tests/foreign_object/models/person.py

```diff
@@ -110,3 +110,28 @@ class Friendship(models.Model):
         related_name="to_friend",
         on_delete=models.CASCADE,
     )
+
+
+class EmployeeProfile(models.Model):
+    employee_country = models.ForeignKey(Country, models.CASCADE, related_name="+")
+    employee_id = models.IntegerField()
+    employee = models.ForeignObject(
+        Person,
+        from_fields=["employee_id", "employee_country"],
+        to_fields=["id", "person_country_id"],
+        related_name="+",
+        on_delete=models.CASCADE,
+    )
+
+    emergency_contact_country = models.ForeignKey(
+        Country, models.CASCADE, related_name="+", null=True
+    )
+    emergency_contact_id = models.IntegerField(null=True)
+    emergency_contact = models.ForeignObject(
+        Person,
+        from_fields=["emergency_contact_id", "emergency_contact_country"],
+        to_fields=["id", "person_country_id"],
+        on_delete=models.SET_NULL,
+        related_name="+",
+        null=True,
+    )
```

Comment:
I think you can avoid having to create a new model here, something like:
```diff
--- a/tests/foreign_object/models/person.py
+++ b/tests/foreign_object/models/person.py
@@ -49,10 +49,10 @@ class Group(models.Model):
 
 class Membership(models.Model):
     # Table Column Fields
-    membership_country = models.ForeignKey(Country, models.CASCADE)
+    membership_country = models.ForeignKey(Country, models.CASCADE, null=True)
     date_joined = models.DateTimeField(default=datetime.datetime.now)
     invite_reason = models.CharField(max_length=64, null=True)
-    person_id = models.IntegerField()
+    person_id = models.IntegerField(blank=True, null=True)
     group_id = models.IntegerField(blank=True, null=True)
 
     # Relation Fields
@@ -74,7 +74,8 @@ class Membership(models.Model):
 
     def __str__(self):
         group_name = self.group.name if self.group_id else "NULL"
-        return "%s is a member of %s" % (self.person.name, group_name)
+        person_name = self.person.name if self.person_id and self.membership_country else "NULL"
+        return "%s is a member of %s" % (person_name, group_name)
 
 
 class Friendship(models.Model):
diff --git a/tests/foreign_object/tests.py b/tests/foreign_object/tests.py
index e8080457da..37465052db 100644
--- a/tests/foreign_object/tests.py
+++ b/tests/foreign_object/tests.py
@@ -16,7 +16,6 @@ from .models import (
     ArticleTag,
     ArticleTranslation,
     Country,
-    EmployeeProfile,
     Friendship,
     Group,
     Membership,
@@ -516,38 +515,17 @@ class MultiColumnFKTests(TestCase):
         Person.objects.bulk_create(objs, 10)
 
     def test_isnull_lookup(self):
-        m1 = Membership.objects.create(
-            membership_country=self.usa, person=self.bob, group_id=None
-        )
-        m2 = Membership.objects.create(
-            membership_country=self.usa, person=self.bob, group=self.cia
-        )
-        self.assertSequenceEqual(
-            Membership.objects.filter(group__isnull=True),
-            [m1],
-        )
+        m1 = Membership.objects.create(person=self.bob)
+        m2 = Membership.objects.create(person_id=self.bob.id)
+        m3 = Membership.objects.create(membership_country=self.usa)
+        m4 = Membership.objects.create()
         self.assertSequenceEqual(
-            Membership.objects.filter(group__isnull=False),
-            [m2],
-        )
-
-        e1 = EmployeeProfile.objects.create(employee=self.bob)
-        e2 = EmployeeProfile.objects.create(
-            employee=self.jim, emergency_contact=self.bob
-        )
-        e3 = EmployeeProfile.objects.create(
-            employee=self.george, emergency_contact_country=self.usa
-        )
-        e4 = EmployeeProfile.objects.create(
-            employee=self.jane, emergency_contact_id=self.bob.id
-        )
-        self.assertSequenceEqual(
-            EmployeeProfile.objects.filter(emergency_contact__isnull=True),
-            [e1, e3, e4],
+            Membership.objects.filter(person__isnull=True),
+            [m2, m3, m4],
         )
         self.assertSequenceEqual(
-            EmployeeProfile.objects.filter(emergency_contact__isnull=False),
-            [e2],
+            Membership.objects.filter(person__isnull=False),
+            [m1],
         )
 
```

Reply:
Ah yeah sure! thanks

Reply:
I adjusted.